linode/manager
# fix: [M3-7249] - Linode Landing flickering

## Description 📝
- Fixes flickering on the Linodes landing page caused by the `PreferenceToggle` component
- The rewrites the `PreferenceToggle` to **not** store duplicate state and rely on React Query for the state preference state directly.

## Changes  🔄
- Removed unused `localStorageKey` prop
- No longer blocks `PreferenceToggle.tsx`'s children from rendering if a preference value is 
## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115251059/b1de96a5-6455-4145-b649-04aba5e37205" /> | <video src="https://github.com/linode/manager/assets/115251059/a6b34fca-913d-4007-9bfc-d3ab2b199d18" /> |

## How to test 🧪

### Reproduction steps
- Click the Akamai logo while on the Linode Landing page to see the flickering

### Verification steps 
- Verify that the flickering no longer happens when you click on the Akamai logo
- Verify that preferences are still loaded and retained normally (for example, if you group by tag and reload, it should retain the group by tag setting)
- Go to http://localhost:3000/profile/settings?preferenceEditor=true and clear your preferences (`{}`)
- Verify that no flickering happens even when your preferences are empty

## As an Author I have considered 🤔
